liquidhaskell-boot/src/Language/Haskell/Liquid/Constraint/Env.hs

```diff
@@ -158,6 +159,14 @@ addBind l x r = do
   put          $ st { binds = bs' } { bindSpans = M.insert i l (bindSpans st) }
   return ((x, F.sr_sort r), {- traceShow ("addBind: " ++ showpp x) -} i)
 
+addRewrites :: F.LocalRewrites -> CG ()
+addRewrites rws = do 
+  st <- get
+  let bid = F.bindEnvSize $ binds st
+  put $ st { localRewrites = F.insertRewrites bid rws $ localRewrites st }
```

Comment:
Should this be `bid-1`? That would be the id of the last inserted binding at least.

Reply:
Can you have `-` inside an identifier?

Reply:
I don't think so, but the suggestion is not to rename bid, but to subtract 1 from the size of the binding environment to get the id of the last inserted bind.

Reply:
But then we need to add 1 again

Reply:
Evidently, I don't get yet what this code is meant to do. So let me ask, why is that we would need to add 1 to the id of the last inserted binding?

Reply:
Because first we are adding the rewrite and then the binding, the local rewrites are attached to the binding inserted at line 878